packages/optimizer/lib/DomTransformer.js

```diff
@@ -168,9 +174,28 @@ class DomTransformer {
    * @return {string} - the transformed html string
    */
   async transformHtml(html, params) {
-    const tree = await treeParser.parse(html);
-    await this.transformTree(tree, params);
-    return treeParser.serialize(tree);
+    async function transform() {
+      const tree = await this.doProfile("parsing", () => treeParser.parse(html));
+
+      await this.doProfile('transform', () => this.transformTree(tree, params));
+
+      return treeParser.serialize(tree);
+    }
+
+    return await this.doProfile('overall', () => transform.call(this));
+  }
+
+  async doProfile(name, f) {
+    if (this.config.profile) {
```

Comment:
nit: this can be _slightly_ cleaner with an early return. that would allow you to also reduce a level of nesting.

```suggestion
    if (!this.config.profile) { return f(); }
    // rest of logic
```